src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -108,6 +108,14 @@ public Task StartReceive(CancellationToken cancellationToken = default)
             return Task.CompletedTask;
         }
 
+        public async Task ChangeConcurrency(PushRuntimeSettings limitations,
+            CancellationToken cancellationToken = new CancellationToken())
+        {
+            await StopReceive(cancellationToken).ConfigureAwait(false);
+            maxConcurrency = limitations.MaxConcurrency;
+            await StartReceive(cancellationToken).ConfigureAwait(false);
```

Comment:
I think this may be better done as the semaphore drain that @SzymonPobiega did in SqlT. I'm not sure if calling `ConnectToBroker` multiple times is problematic or not. I don't know too much so will leave it to your judgment though.

Reply:
We talked about it with @WojcikMike and decided to give it a try tomorrow during a pairing session.